src/main/java/net/greghaines/jesque/worker/WorkerImpl.java

```diff
@@ -529,21 +533,40 @@ protected void process(final Job job, final String curQueue) {
             final Object instance = this.jobFactory.materializeJob(job);
             final Object result = execute(job, curQueue, instance);
             success(job, instance, result, curQueue);
+        } catch (DontPerformException e) {
+            try {
+                Thread.sleep(dontPerformSleepTime);
```

Comment:
If the "dont perform" job is the only one left in all queues, the sleep is needed, because otherwise Jesque will try to immediately execute the job over and over again. 

For my use case (preventing the concurrent execution of two jobs of the same kind) this would be the case, if one job is executing and the only other job in all queues is of the same kind.

https://github.com/bartolsthoorn/resque-workers-lock does this by itself re-enqueuing the job. Maybe that is the better solution?